src/zenml/utils/source_utils.py

```diff
@@ -457,11 +457,22 @@ def import_python_file(file_path: str) -> types.ModuleType:
         file_path: Path to python file that should be imported.
 
     Returns:
-        The imported module.
+        imported module: Module
     """
     # Add directory of python file to PYTHONPATH so we can import it
     file_path = os.path.abspath(file_path)
     sys.path.append(os.path.dirname(file_path))
-
     module_name = os.path.splitext(os.path.basename(file_path))[0]
-    return importlib.import_module(module_name)
+
+    # In case the module is already fully or partially imported and the module
+    #  path is something like materializer.materializer the full path needs to
+    #  be checked for in the sys.modules to avoid getting an empty namespace
+    #  module
+    full_module_path = os.path.splitext(
```

Comment:
I would unit test this part of the code, seems very fragile! :-D

Reply:
Absolutely agree - now that we scoped this for next release there will definitely be enough time for that.

Reply:
the function in source_utils and the cli pipeline run functions got some unit-test-love. Have a look and let me know if this should cover it (one unit test still runs into I/O Errors on cleanup, will be looking at that in the meantime)